lib/Lift/JumpTargetManager.cpp

```diff
@@ -528,6 +528,9 @@ JumpTargetManager::JumpTargetManager(Function *TheFunction,
   StringMap<cl::Option *> &Options(cl::getRegisteredOptions());
   getOption<bool>(Options, "enable-load-pre")->setInitialValue(false);
   getOption<unsigned>(Options, "memdep-block-scan-limit")->setInitialValue(100);
+  const char *EarlyCSEOption = "earlycse-mssa-optimization-cap";
+  getOption<unsigned>(Options, EarlyCSEOption)->setInitialValue(2000);
```

Comment:
Some recent changes to `LICM` aimed to improve `MemorySSA` (https://reviews.llvm.org/D121987 and https://reviews.llvm.org/D121381) by delaying some actions/optimizations in that Pass. By doing that we are loosing accuracy, since there are revng tests relying on `MemorySSA` a lot. In order to still use those optimizations, but also have our tests pass we need to increase the Cap of the clobbering calls in `EarlyCSE`, since they introduced that cl::opt with https://reviews.llvm.org/D58248, to even pay the price the EarlyCSE is not powerful enough when adjusting the number of calls to `getClobberingMemoryAccess`.
An alternative would be to run GVN, but it is very slow.

I will add a comment.